sky/core.py

```diff
@@ -953,6 +956,99 @@ def storage_ls() -> List[Dict[str, Any]]:
+        # check if anything in state.db doesn't exist in the external state
+        only_in_internal_state = internal_buckets_status[storetype].difference(
+            external_buckets_status[storetype])
+        # remove storages that exist in state.db but not in external state
+        for s_name in only_in_internal_state:
+            global_user_state.remove_storage(s_name, storetype)
+            sky_logging.print(f'{red}Removed{reset} '
+                              f'{bold}{storetype.value}{reset} '
+                              f'bucket: {s_name}')
+        # check if anything in external state doesn't exist in state.db
+        only_in_external_state = external_buckets_status[storetype].difference(
+            internal_buckets_status[storetype])
+        # add storage that exist in external state but not in state.db
+        for s_name in only_in_external_state:
+            # Storage with S_NAME is already in internal_state from different
+            # cloud storage provider. Obtain StorageMetedata to update handle
+            for s_type in internal_buckets_status.keys():
+                if s_type != storetype and s_name in internal_buckets_status[
+                        s_type]:
+                    handle = \
+                        global_user_state.get_handle_from_storage_name(s_name)
+                    if handle is not None:
+                        region = storage_lib.get_bucket_region(
+                            s_name, storetype)
+                        store_class = \
+                            storage_lib.get_abstract_store_from_storetype(
+                            storetype)
+                        handle.sky_stores[
+                            storetype] = store_class.StoreMetadata(
+                                name=s_name,
+                                source=None,
+                                region=region,
+                                is_sky_managed=True)
+                        break
+            # Storage with s_name created for the first time
+            else:
+                store_class = storage_lib.get_abstract_store_from_storetype(
+                    storetype)
+                region = storage_lib.get_bucket_region(s_name, storetype)
+                store_metadata = store_class.StoreMetadata(name=s_name,
+                                                           source=None,
+                                                           region=region,
+                                                           is_sky_managed=True)
+                handle = storage_lib.Storage.StorageMetadata(
+                    storage_name=s_name,
+                    source=None,
+                    sky_stores={storetype: store_metadata})
+            if handle is not None:
+                # update the externally created storage
+                global_user_state.add_or_update_storage(
+                    s_name, handle, global_user_state.StorageStatus.READY)
```

Comment:
Wondering - can we not directly instantiate a `Storage` object here with `s_name` and call add_store on it to add it to local state?

Reply:
Great point. Confirmed that it works with instantiating `Storage` object and calling `add_store` for both cases of adding it to the local state for the first time and non-first time. `StoreType.to_store` isn't used anymore with this change, but wondering if we should still maintain it since it should be useful?